numba/np/new_arraymath.py

```diff
@@ -244,9 +244,9 @@ def inner(arr, axis):
     return inner
 
 
-@lower_builtin(np.sum, types.Array, types.np_intp, types.DTypeSpec)
+@lower_builtin(np.sum, types.Array, types.Integer, types.DTypeSpec)
 @lower_builtin(np.sum, types.Array, types.IntegerLiteral, types.DTypeSpec)
-@lower_builtin("array.sum", types.Array, types.np_intp, types.DTypeSpec)
+@lower_builtin("array.sum", types.Array, types.Integer, types.DTypeSpec)
```

Comment:
This is generalizing the the axis argument to accept any kind of integers. If we are doing that, I think the implementation should normalize it to `np_intp`. So when this calls 

```
context.compile_internal(builder, array_sum_impl_axis, sig, args)
```
The `sig.args[1]` should be `np_intp`.